src/test/java/org/broadinstitute/hellbender/IntelInflaterDeflaterIntegrationTest.java

```diff
@@ -0,0 +1,148 @@
+        args.add("--use_jdk_inflater"); args.add(String.valueOf(use_jdk_inflater));
+        args.add("--use_jdk_deflater"); args.add(String.valueOf(use_jdk_deflater));
+
+        // store current default factories, so they can be restored later
+        InflaterFactory currentInflaterFactory = BlockGunzipper.getDefaultInflaterFactory();
+        DeflaterFactory currentDeflaterFactory = BlockCompressedOutputStream.getDefaultDeflaterFactory();
+
+        // set default factories to jdk version
+        // because PrintReads cannot change the factory to Jdk if it was already set to Intel
+        BlockGunzipper.setDefaultInflaterFactory(new InflaterFactory());
+        BlockCompressedOutputStream.setDefaultDeflaterFactory(new DeflaterFactory());
+
+        // run PrintReads
+        runCommandLine(args);
+
+        // restore default factories
+        BlockGunzipper.setDefaultInflaterFactory(currentInflaterFactory);
+        BlockCompressedOutputStream.setDefaultDeflaterFactory(currentDeflaterFactory);
+
+        // validate input and output files are the same
+        SamAssertionUtils.assertSamsEqual(outFile, ORIG_BAM);
+    }
+
+    @Test
+    public void deflateInflateWithIntel() throws DataFormatException {
+        if (!isIntelInflaterDeflaterSupported()) {
+            throw new SkipException("IntelInflater/IntelDeflater not available on this platform");
+        }
+
+        // create buffers and random input
+        final int LEN = 64 * 1024;
+        final byte[] input = new RandomDNA().nextBases(LEN);
+        final byte[] compressed = new byte[2 * LEN];
+        final byte[] result = new byte[LEN];
+
+        final IntelInflaterFactory intelInflaterFactory = new IntelInflaterFactory();
+        final IntelDeflaterFactory intelDeflaterFactory = new IntelDeflaterFactory();
+        Assert.assertTrue(intelInflaterFactory.usingIntelInflater());
+        Assert.assertTrue(intelDeflaterFactory.usingIntelDeflater());
+
+        for (int i = 0; i < 10; i++) {
+            // create deflater with compression level i
+            final Deflater deflater = intelDeflaterFactory.makeDeflater(i, true);
+
+            // setup deflater
+            deflater.setInput(input);
+            deflater.finish();
+
+            // compress data
+            int compressedBytes = 0;
+
+            // buffer for compressed data is 2x the size of the data we are compressing
+            // so this loop should always finish in one iteration
+            while (!deflater.finished()) {
+                compressedBytes = deflater.deflate(compressed, 0, compressed.length);
+            }
+            deflater.end();
+
+            // decompress and check output == input
+            Inflater inflater = intelInflaterFactory.makeInflater(true);
+
+            inflater.setInput(compressed, 0, compressedBytes);
+            inflater.inflate(result);
+            inflater.end();
+
+            Assert.assertEquals(input, result);
```

Comment:
Yes, just to make sure the test doesn't pass somehow using previous results. Done.